src/screens/Profile.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from "react";
+import { withStore } from "../assets/helpers/store";
+import { disableBodyScroll, enableBodyScroll, clearAllBodyScrollLocks } from 'body-scroll-lock';
+import Modal from "../components/Modals/LoginSignUpRedirectModal";
+
+class Profile extends Component {
+   constructor(props) {
+      super(props);
+      this.state = {
+         viewOnly: true,
+         selectedCategory: '',
+         events: [],
+         loggedInUserName: 'User',
+         userLoggedIn: 'false',
+         loggedInUser: {},
+         tggle: false,
+      }
+   }
+
+
+   componentDidMount() {
+      if (this.props.store.loggedInUser == null) {
+         disableBodyScroll(this.targetElement);
+         this.setState({
+            isShowing: true
+         });
+      } else {
+         this.setState({ loggedInUser: this.props.store.loggedInUser })
+         this.setState({ loggedInUserName: this.props.store.loggedInUser.userFirstName })
+      }
+   }
+
+   openModalHandler = () => {
+      disableBodyScroll(this.targetElement);
+      this.setState({
+         isShowing: true
```

Comment:
We should be more specific as to what `isShowing` means. I'm thinking `modalIsShowing`